### PR TITLE
add endpoint to fetch directors related information

### PR DESCRIPTION
**Pull request by swalke-ch**
add endpoint to fetch directors related information

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/acsp/service/OfficerService.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    private OfficersApi retrieveOfficerAppointments(String companyNumber, ApiClient apiClient, Integer startIndex, Integer itemsPerPage)
+            throws ServiceException {
+        String uri = GET_OFFICERS_URI.expand(companyNumber).toString();
+        OfficersApi officersApi = null;
+        try {
+            OfficersList officersList = apiClient.officers().list(uri);
+            officersList.addQueryParams("items_per_page", itemsPerPage.toString());
+            officersList.addQueryParams("start_index", startIndex.toString());
+            officersList.addQueryParams("register_type", "directors");
```
Comment (sampankumar) [inline src/main/java/uk/gov/companieshouse/acsp/service/OfficerService.java:74]: We can add register_type as query parameter in our controller.

File: src/main/java/uk/gov/companieshouse/acsp/service/OfficerService.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    private static final UriTemplate GET_OFFICERS_URI =
+        new UriTemplate("/company/{companyNumber}/officers");
+
+    public OfficersApi getOfficers(String passThroughHeader, String companyNumber) throws ServiceException {
+        try {
+            ApiClient apiClient = apiClientService.getApiClient(passThroughHeader);
+
+            OfficersApi officersApi;
+            int startIndex = 0;
```
Comment (sampankumar) [inline src/main/java/uk/gov/companieshouse/acsp/service/OfficerService.java:41]: Lets use constants for these

File: src/main/java/uk/gov/companieshouse/acsp/service/OfficerService.java (hunk 1 above)
Comment (swalke-ch) [reply src/main/java/uk/gov/companieshouse/acsp/service/OfficerService.java]: done

File: src/main/java/uk/gov/companieshouse/acsp/service/OfficerService.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    private OfficersApi retrieveOfficerAppointments(String companyNumber, ApiClient apiClient, Integer startIndex, Integer itemsPerPage)
+            throws ServiceException {
+        String uri = GET_OFFICERS_URI.expand(companyNumber).toString();
+        OfficersApi officersApi = null;
+        try {
+            OfficersList officersList = apiClient.officers().list(uri);
+            officersList.addQueryParams("items_per_page", itemsPerPage.toString());
+            officersList.addQueryParams("start_index", startIndex.toString());
+            officersList.addQueryParams("register_type", "directors");
+            officersList.addQueryParams("register_view", "true");
```
[the remark this answers is not in the document]
Comment (swalke-ch) [reply src/main/java/uk/gov/companieshouse/acsp/service/OfficerService.java]: done

File: src/main/java/uk/gov/companieshouse/acsp/service/OfficerService.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+import java.io.IOException;
+
+import static uk.gov.companieshouse.acsp.AcspApplication.APP_NAMESPACE;
+import static uk.gov.companieshouse.acsp.util.Constants.TRUE;
+
+@Service
+public class OfficerService {
+
+    private final ApiClientService apiClientService;
```
Comment (sampankumar) [inline src/main/java/uk/gov/companieshouse/acsp/service/OfficerService.java:26]: does this need to be autowired?

Comment (swalke-ch) [reply src/main/java/uk/gov/companieshouse/acsp/service/OfficerService.java:26]: Yes, It  is autowired using constructor injection. Line 33

**Review by sampankumar — APPROVED** [review]

**Review by sardhani-ch — APPROVED** [review]

**Pull request merged**